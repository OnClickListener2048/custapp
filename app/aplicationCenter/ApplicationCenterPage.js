/**
 * Created by jinglan on 2017/6/7.
 */
import React, {Component} from 'react';
import {Dimensions, TouchableOpacity,
    DeviceEventEmitter,
    InteractionManager,Platform} from 'react-native';
import {loadOutSourceCount} from "../apis/outSource";
import {
    Text,
    Image,
    View,
} from 'react-native';
import CommunalNavBar from '../main/GDCommunalNavBar';

import TopcenterImgBottomTitleView from '../view/TopcenterImgBottomTitleView';
import styles from './css/ApplictionCenterPageStyle'
const window = Dimensions.get('window');
import MyOutSideWorkPage from "../outWork/myOutSideWork/MyOutSideWorkPage";
import Toast from 'react-native-root-toast';
import ScrollViewTop from "./scrollViewTop";

export const SCREEN_WIDTH = window.width;
import errorText from '../util/ErrorMsg';


/** 应用 */
export default class ApplicationCenterPage extends Component{
    static navigatorStyle = {
        navBarHidden: true, // 隐藏默认的顶部导航栏
    };
    constructor(props) {
        super(props);

        this.state = {
            bdNum:null,
            canClickBtn : false,
        };


        this._loadCount = this._loadCount.bind(this);
        this.props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
    }

    onNavigatorEvent(event) { // this is the onPress handler for the two buttons together
        // console.log('ApplicationCenterPage event.type', event.type);
        console.log('ApplicationCenterPage event', JSON.stringify(event));
        if(event.id==='willAppear'){
            this._loadCount();
            this.state.canClickBtn = true;

        }
    }
    static renderTitleItem() {
        return(
            <Text style={[styles.navbarTitleItemStyle,{fontSize:18,color:'#323232'}]}>应用中心</Text>
        );
    }

    toMyOutSideWork(){

        if (this.state.canClickBtn === false){
            return;
        }

        this.state.canClickBtn = false;


        this.props.navigator.push({
                screen: 'MyOutSideWorkPage',
                backButtonTitle: '返回', // 返回按钮的文字 (可选)
                backButtonHidden: false, // 是否隐藏返回按钮 (可选)
                title:'我的外勤',

            });
    }

    componentDidMount() {
        this.subscription = DeviceEventEmitter.addListener('loginSuccess', (data) => {
            console.log('应用 loginSuccess');
            try {
                this._loadCount();
            } catch (e) {
                console.log(e + "");
            }
        });
    }

    componentWillUnmount() {
        // 销毁
        this.subscription.remove();
    }

    componentWillMount() {
        this._loadCount();
    }

    //获取每个外勤状态数量
    _loadCount(){

        // let loading  = SActivityIndicator.show(true, "加载中...");
        loadOutSourceCount().then(

            (responseData) => {
                // SActivityIndicator.hide(loading);
                if(responseData !== null && responseData.data !== null) {
                    this.outSourceCountObj = {};
                    console.log("开始请求2是"+responseData.data.todoNum+"，"+responseData.data.totalNum+"，"+responseData.data.inProgressNum);

                    this.setState({
                        bdNum:responseData.data.todoNum+responseData.data.inProgressNum,
                    });

                    if(this.refs.myoutItem) {
                        this.refs.myoutItem._setBageNum(this.state.bdNum);
                    }

                    this.props.navigator.setTabBadge({
                        badge: this.state.bdNum <= 0 ? null : this.state.bdNum // 数字气泡提示, 设置为null会删除
                    });

                }
            },
            (e) => {
                    // SActivityIndicator.hide(loading);
                console.log("获取失败" , JSON.stringify(e));
                Toast.show(errorText( e ));
            },
        );

    }



    render() {
        console.log("应用中心render="+this.state.bdNum);
        return(
        <View style={styles.container}>

            <CommunalNavBar

                titleItem = {() => ApplicationCenterPage.renderTitleItem()}
            />
            <ScrollViewTop/>

            <View style={styles.applicationViewContainer}>

                <TouchableOpacity
                    onPress={() => {this.toMyOutSideWork()}}
                    style={{ marginLeft: 15,marginTop: 15, height: 100, width: (SCREEN_WIDTH - 45)/2,}}

                >
                    <TopcenterImgBottomTitleView ref="myoutItem"
                                                 applicationTitle='我的外勤'
                                                 applicationImg = {require('../img/field.png')}
                                                 style={{height: 100, width: (SCREEN_WIDTH - 45)/2,}}
                                                 textStyle={{color: '#ef0c35',  alignSelf: 'flex-end'}}
                                                 badge={this.state.bdNum}
                    />
                </TouchableOpacity>
                <TouchableOpacity
                    // onPress={() => {this.toDialog()}}
                    style={{ marginLeft: 15,marginTop: 15, height: 100, width: (SCREEN_WIDTH - 45)/2,}}

                >
                <TopcenterImgBottomTitleView applicationTitle='CRM'
                                             applicationImg = {require('../img/crm_h.png')}
                                             style={{height: 100, width: (SCREEN_WIDTH - 45)/2,}}
                                             textStyle={{color: '#ef0c35',  alignSelf: 'flex-end'}}
                                             badge={0}
                />
                </TouchableOpacity>
                <TouchableOpacity
                    // onPress={() => {this.toMyOutSideWork()}}
                    style={{ marginLeft: 15,marginTop: 15, height: 100, width: (SCREEN_WIDTH - 45)/2,}}

                >
                <TopcenterImgBottomTitleView applicationTitle='工作统计'
                                             applicationImg = {require('../img/statistical_h.png')}
                                             style={{height: 100, width: (SCREEN_WIDTH - 45)/2}}
                                             textStyle={{color: '#ef0c35',  alignSelf: 'flex-end'}}
                                             badge={0}
                />
            </TouchableOpacity>
                <TouchableOpacity
                    // onPress={() => {this.toMyOutSideWork()}}
                    style={{ marginLeft: 15,marginTop: 15, height: 100, width: (SCREEN_WIDTH - 45)/2,}}

                >
                <TopcenterImgBottomTitleView applicationTitle='工作日志'
                                             applicationImg = {require('../img/log_h.png')}
                                             style={{height: 100, width: (SCREEN_WIDTH - 45)/2,}}
                                             textStyle={{color: '#ef0c35',  alignSelf: 'center'}}
                                             badge={0}
                />
                </TouchableOpacity>

            </View>

        </View>
        );
    }
}