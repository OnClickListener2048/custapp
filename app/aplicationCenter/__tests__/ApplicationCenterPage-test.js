/**
 * Created by zhuangzihao on 2017/8/22.
 */
import 'react-native';
import React from 'react';
import fetchMock from 'fetch-mock';
import {shallow} from 'enzyme';
import ApplicationCenterPage from '../ApplicationCenterPage';
import responseData from './responseData.json'
import renderer from 'react-test-renderer';

const navigator = Object.create(null);
navigator.setOnNavigatorEvent = jest.fn();
const wrapper = shallow(
    <ApplicationCenterPage navigator={navigator}/>
);
let instance = wrapper.instance();
fetchMock.post('*', responseData);
test('test _loadCount 测试消息数是否正确渲染', (done) => {
    instance.componentWillMount()
    jest.useRealTimers();
    setTimeout(function() {
        expect(instance.state.bdNum).toEqual(responseData.data.todoNum+responseData.data.inProgressNum);
        done();
    }, 1000);

})

test('快照正确渲染', () => {
    const tree = renderer.create(
        <ApplicationCenterPage  navigator={navigator}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    // 只能操作属性
    expect(tree).toMatchSnapshot();
});

test('test toMyOutSideWork 外勤按钮重复点击', () => {
    instance.toMyOutSideWork()
    expect(instance.state.canClickBtn).toEqual(false);
});