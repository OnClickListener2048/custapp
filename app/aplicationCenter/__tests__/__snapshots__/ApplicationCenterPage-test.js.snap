// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照正确渲染 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderBottomColor": "#DCDCDC",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingTop": 15,
        "width": 750,
      }
    }
  >
    <View />
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            Object {
              "color": "#323232",
              "fontSize": 18,
            },
          ]
        }
      >
        应用中心
      </Text>
    </View>
    <View />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#dddddd",
      }
    }
  >
    <RCTScrollView
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      pagingEnabled={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <Image
          source={1}
          style={
            Object {
              "flex": 1,
              "width": 750,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": 250,
        "width": 750,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
          "width": 352.5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "height": 44,
              "justifyContent": "center",
              "width": 70,
            },
            Object {
              "height": 100,
              "width": 352.5,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "justifyContent": "center",
                "resizeMode": "stretch",
              },
              Object {
                "height": 100,
                "width": 352.5,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginBottom": 2,
                  "resizeMode": "contain",
                },
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 70,
                "position": "absolute",
                "width": 130,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "fontSize": 12,
                  "marginTop": 10,
                },
              ]
            }
            textAlign="center"
          >
            我的外勤
          </Text>
        </Image>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
          "width": 352.5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "height": 44,
              "justifyContent": "center",
              "width": 70,
            },
            Object {
              "height": 100,
              "width": 352.5,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "justifyContent": "center",
                "resizeMode": "stretch",
              },
              Object {
                "height": 100,
                "width": 352.5,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginBottom": 2,
                  "resizeMode": "contain",
                },
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 70,
                "position": "absolute",
                "width": 130,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "fontSize": 12,
                  "marginTop": 10,
                },
              ]
            }
            textAlign="center"
          >
            CRM
          </Text>
        </Image>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
          "width": 352.5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "height": 44,
              "justifyContent": "center",
              "width": 70,
            },
            Object {
              "height": 100,
              "width": 352.5,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "justifyContent": "center",
                "resizeMode": "stretch",
              },
              Object {
                "height": 100,
                "width": 352.5,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginBottom": 2,
                  "resizeMode": "contain",
                },
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 70,
                "position": "absolute",
                "width": 130,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "fontSize": 12,
                  "marginTop": 10,
                },
              ]
            }
            textAlign="center"
          >
            工作统计
          </Text>
        </Image>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
          "width": 352.5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "height": 44,
              "justifyContent": "center",
              "width": 70,
            },
            Object {
              "height": 100,
              "width": 352.5,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "justifyContent": "center",
                "resizeMode": "stretch",
              },
              Object {
                "height": 100,
                "width": 352.5,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginBottom": 2,
                  "resizeMode": "contain",
                },
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 70,
                "position": "absolute",
                "width": 130,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "fontSize": 12,
                  "marginTop": 10,
                },
              ]
            }
            textAlign="center"
          >
            工作日志
          </Text>
        </Image>
      </View>
    </View>
  </View>
</View>
`;

exports[`快照正确渲染 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderBottomColor": "#DCDCDC",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingTop": 15,
        "width": 750,
      }
    }
  >
    <View />
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            Object {
              "color": "#323232",
              "fontSize": 18,
            },
          ]
        }
      >
        应用中心
      </Text>
    </View>
    <View />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#dddddd",
      }
    }
  >
    <RCTScrollView
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      pagingEnabled={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <Image
          source={1}
          style={
            Object {
              "flex": 1,
              "width": 750,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": 250,
        "width": 750,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
          "width": 352.5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "height": 44,
              "justifyContent": "center",
              "width": 70,
            },
            Object {
              "height": 100,
              "width": 352.5,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "justifyContent": "center",
                "resizeMode": "stretch",
              },
              Object {
                "height": 100,
                "width": 352.5,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginBottom": 2,
                  "resizeMode": "contain",
                },
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 70,
                "position": "absolute",
                "width": 130,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "fontSize": 12,
                  "marginTop": 10,
                },
              ]
            }
            textAlign="center"
          >
            我的外勤
          </Text>
        </Image>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
          "width": 352.5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "height": 44,
              "justifyContent": "center",
              "width": 70,
            },
            Object {
              "height": 100,
              "width": 352.5,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "justifyContent": "center",
                "resizeMode": "stretch",
              },
              Object {
                "height": 100,
                "width": 352.5,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginBottom": 2,
                  "resizeMode": "contain",
                },
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 70,
                "position": "absolute",
                "width": 130,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "fontSize": 12,
                  "marginTop": 10,
                },
              ]
            }
            textAlign="center"
          >
            CRM
          </Text>
        </Image>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
          "width": 352.5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "height": 44,
              "justifyContent": "center",
              "width": 70,
            },
            Object {
              "height": 100,
              "width": 352.5,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "justifyContent": "center",
                "resizeMode": "stretch",
              },
              Object {
                "height": 100,
                "width": 352.5,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginBottom": 2,
                  "resizeMode": "contain",
                },
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 70,
                "position": "absolute",
                "width": 130,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "fontSize": 12,
                  "marginTop": 10,
                },
              ]
            }
            textAlign="center"
          >
            工作统计
          </Text>
        </Image>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100,
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
          "width": 352.5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "height": 44,
              "justifyContent": "center",
              "width": 70,
            },
            Object {
              "height": 100,
              "width": 352.5,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "justifyContent": "center",
                "resizeMode": "stretch",
              },
              Object {
                "height": 100,
                "width": 352.5,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginBottom": 2,
                  "resizeMode": "contain",
                },
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 70,
                "position": "absolute",
                "width": 130,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "fontSize": 12,
                  "marginTop": 10,
                },
              ]
            }
            textAlign="center"
          >
            工作日志
          </Text>
        </Image>
      </View>
    </View>
  </View>
</View>
`;
